lib/everypolitician_extensions.rb

```diff
@@ -52,6 +58,32 @@ def top_identifiers
                            .sort_by { |s, ids| [-ids.size, s] }
                            .map { |s, _ids| s }
     end
+
+    private
+
+    def current_cabinet_memberships
+      all_cabinet_memberships.select do |mem|
+        !mem.start_date.nil? && (mem.end_date.to_s.empty? || mem.end_date >= start_date.to_s)
```

Comment:
I've been peering at this for quite a while now, and it's making my brain hurt. 

I can't quite tell whether `!mem.start_date.nil?` is meant to be doing something different from a plain `mem.start_date`, nor what the `&&` condition is actually checking for.

I suspect this might be clearer as a `reject`, or possibly even more than one `reject`, rather than a `select`, but it'd almost certainly be even clearer still moving these to named methods on  `CabinetMembership`.